hy/cmdline.py

```diff
@@ -73,25 +75,35 @@ def __call__(self, code=None):
 builtins.exit = HyQuitter('exit')
 
 
-def print_python_code(_ast):
-    # astor cannot handle ast.Interactive, so disguise it as a module
-    _ast_for_print = ast.Module()
-    _ast_for_print.body = _ast.body
-    print(astor.codegen.to_source(_ast_for_print))
-
-
 class HyREPL(code.InteractiveConsole):
-    def __init__(self, spy=False, locals=None, filename="<input>"):
+    def __init__(self, spy=False, output_fn=None, locals=None,
+                 filename="<input>"):
+
         self.spy = spy
+
+        if output_fn is None:
+            self.output_fn = repr
+        elif callable(output_fn):
+            self.output_fn = output_fn
+        else:
+            f = hy_symbol_mangle(output_fn)
+            if "." in output_fn:
+                module, f = f.rsplit(".", 1)
+                self.output_fn = getattr(importlib.import_module(module), f)
+            else:
+                self.output_fn = __builtins__[f]
+
         code.InteractiveConsole.__init__(self, locals=locals,
                                          filename=filename)
 
     def runsource(self, source, filename='<input>', symbol='single'):
         global SIMPLE_TRACEBACKS
         try:
-            tokens = tokenize(source)
-        except PrematureEndOfInput:
-            return True
+            try:
+                tokens = tokenize(source)
+            except PrematureEndOfInput:
+                return True
+            tokens = tokenize("(do " + source + "\n)")
```

Comment:
In the current implementation, `(do ...)` already doesn't appear. (Your example above no longer has `do` in its output.)